src/utils/DockerInstallStatusProvider.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AsyncLazy } from "./lazy";
+import { execAsync } from "./spawnAsync";
+
+export interface IDockerInstallStatusProvider {
+    isDockerInstalled(): Promise<boolean>
+}
+
+class DockerInstallStatusProvider implements IDockerInstallStatusProvider {
```

Comment:
Although it's unlikely the user would try, perhaps we should block the `installDocker` command by checking if Docker is already installed using this.

Reply:
Added this check. Note that I am not blocking the installation, instead warning the user and it is user's choice.